lib/graph.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class Graph
+  attr_reader :klass, :g
+
+  def initialize(klass)
+    @klass = klass
+    @g = GraphViz.new(:G, type: :digraph)
+
+    add_nodes
+    add_edges
+    add_start_node
+  end
+
+  private
+
+  def add_nodes
+    klass.states.each { |s| g.add_nodes(s.name.to_s) }
```

Comment:
is there any reason to do `to_s` here? better make all sanitization in setter/getter methods.